scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -948,6 +921,78 @@ public List<FilteredTopWorkflow> getTopWorkflowsWithIssues(int numberOfWorkflows
         });
     }
 
+    public List<FilteredTopWorkflowExecutionTime> getTopExecutionTimeWorkflows(int numberOfWorkflows,
+            final String workflowName, String user, String tenant, final long startTime, final long endTime) {
+
+        return executeReadOnlyTransaction(session -> {
+
+            String selectQuery = "select jobName, projectName, avg(finishedTime - startTime) as executionTime, count(*) as numberOfExecution from JobData where startTime > 0 and finishedTime > 0 ";
+            String groupByQuery = "group by jobName, projectName order by executionTime desc";
+            Query query = getTopWorkflowsQuery(session,
+                                               workflowName,
+                                               user,
+                                               tenant,
+                                               startTime,
+                                               endTime,
+                                               numberOfWorkflows,
+                                               selectQuery,
+                                               groupByQuery);
+
+            List<Object[]> list = query.list();
+            return list.stream()
+                       .map(nameAndCount -> new FilteredTopWorkflowExecutionTime(nameAndCount[0].toString(),
+                                                                                 nameAndCount[1].toString(),
+                                                                                 Double.parseDouble(nameAndCount[2].toString()),
+                                                                                 Long.parseLong(nameAndCount[3].toString())))
+                       .collect(Collectors.toList());
+        });
+    }
+
+    private Query getTopWorkflowsQuery(Session session, String workflowName, String user, String tenant, long startTime,
+            long endTime, int numberOfWorkflows, String selectQuery, String groupByQuery) {
+        StringBuilder queryString = new StringBuilder(selectQuery);
+
+        boolean hasWorkflowName = !Strings.isNullOrEmpty(workflowName);
+        boolean hasUser = !Strings.isNullOrEmpty(user);
+        boolean hasTenant = !Strings.isNullOrEmpty(tenant);
+
+        if (hasWorkflowName) {
+            queryString.append("and jobName = :workflowName ");
+        }
+        if (hasUser) {
+            queryString.append("and owner = :user ");
+            if (hasTenant) {
+                queryString.append("and (tenant = :tenant or tenant = null) ");
+            }
+        }
+        if (startTime != 0) {
+            queryString.append("and finishedTime >= :startTime ");
+        }
+        if (endTime != 0) {
+            queryString.append("and finishedTime <= :endTime ");
+        }
+        queryString.append(groupByQuery);
+
+        Query query = session.createQuery(queryString.toString());
```

Comment:
I don't see the notion of select .... from (select ... from JobData) as something